lib/gui/components/update-notifier/services/update-notifier.js

```diff
@@ -17,104 +17,45 @@
-   * We assume the received latest version number will not increase
-   * while Etcher is running and memoize it
    *
-   * @fulfil {String} - latest version
-   * @returns {Promise}
+   * @param {String} releaseType - release type
+   * @returns {Boolean} whether sleep is allowed
    *
-   * @example
-   * UpdateNotifierService.getLatestVersion().then((latestVersion) => {
-   *   console.log(`The latest version is: ${latestVersion}`);
-   * });
-   */
-  this.getLatestVersion = _.memoize(() => {
-    return $q((resolve, reject) => {
-      return etcherLatestVersion((url, callback) => {
-        return $http.get(url).then((response) => {
-          return callback(null, response.data);
-        }).catch((error) => {
-          return callback(error);
-        });
-      }, (error, latestVersion) => {
-        if (error) {
-
-          // The error status equals this number if the request
-          // couldn't be made successfully, for example, because
-          // of a timeout on an unstable network connection.
-          const ERROR_CODE_UNSUCCESSFUL_REQUEST = -1;
-
-          if (error.status === ERROR_CODE_UNSUCCESSFUL_REQUEST) {
-            return resolve(CURRENT_VERSION);
-          }
-
-          return reject(error);
-        }
-
-        return resolve(latestVersion);
-      });
-    });
-
-  // Arbitrary identifier for the memoization function
-  }, _.constant('latest-version'));
-
-  /**
-   * @summary Check if the current version is the latest version
-   * @function
-   * @public
-   *
-   * @fulfil {Boolean} - is latest version
-   * @returns {Promise}
-   *
-   * @example
-   * UpdateNotifierService.isLatestVersion().then((isLatestVersion) => {
-   *   if (!isLatestVersion) {
-   *     console.log('There is an update available');
-   *   }
-   * });
+   * if (shouldAllowSleep(release.RELEASE_TYPE.PRODUCTION)) {
+   *   console.log('We can allow the user to put the update notifier to sleep');
+   * }
    */
-  this.isLatestVersion = () => {
-    return this.getLatestVersion().then((version) => {
-      return semver.gte(CURRENT_VERSION, version);
-    });
+  const shouldAllowSleep = (releaseType) => {
+    return releaseType === release.RELEASE_TYPE.PRODUCTION;
   };
 
   /**
    * @summary Determine if its time to check for updates
    * @function
    * @public
    *
+   * @param {String} releaseType - release type
    * @returns {Boolean} should check for updates
    *
    * @example
-   * if (UpdateNotifierService.shouldCheckForUpdates()) {
+   * if (UpdateNotifierService.shouldCheckForUpdates(release.RELEASE_TYPE.PRODUCTION)) {
    *   console.log('We should check for updates!');
    * }
    */
-  this.shouldCheckForUpdates = () => {
+  this.shouldCheckForUpdates = (releaseType) => {
     const lastUpdateNotify = settings.get('lastUpdateNotify');
 
-    if (!settings.get('sleepUpdateCheck') || !lastUpdateNotify) {
+    if (!settings.get('sleepUpdateCheck') || !lastUpdateNotify || !shouldAllowSleep(releaseType)) {
```

Comment:
Although my gut-feeling says checking a function named `shouldAllowSleep` inside `shouldCheckForUpdates` feels a bit odd ;-)

I agree with what the code is _actually_ doing (i.e. always check for an update if it's a non-release build), it's just the choice of `shouldAllowSleep` function-name that feels a bit strange when used here. IYSWIM.